### ISSUE TITLE
A11y_Azure Tools Storage Explorer_Accessible environment_Desired Storage Explorer theme_NonTextContrast:In System Default/Light theme Color Contrast of focus Indicator for "Search" button failing minimum Color Contrast ratio i.e., 3:1

### ISSUE THREAD

**Issue by ManishaC1** [microsoft/AzureStorageExplorer#5731]
[Check out Accessibility Insights!](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-nikpap%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458847543876&sdata=6wiDneF0tmnVt4CroW43rkfVbB%2BL%2BAAdn0nD7xR5hQM%3D&reserved=0) - Identify accessibility bugs before check-in and make bug fixing faster and easier.

**GitHub Tags:**
#A11yMAS; #A11yTCS; #Win11; #DesktopApp; #A11ySev2; #BM_Azure Tools Storage Explorer_May2022; #Azure Tools Storage Explorer; #WCAG1.4.11; #FTP; #A11yWCAG2.1; #ColorContrast;

## Environment Details:
Application: Microsoft Azure Storage Explore
version: 1.24.0

## OS Details
Microsoft Windows 11 Enterprise
version 21H2 OS Build 22000.613

## Repro Steps:

1. Launch Storage Explorer
2. Navigate to  "View" menu and click on it.
3. Navigate to "Themes" and click on it 
4. Select System Default/Light theme.
5. Focus "Search" button  present in the "Explorer" pane and above the tree view list.
6. Use Color Contrast Tool to check the color contrast ratio for Visible focus.

## Actual results:
"Search" button visible focus is less than 3:1. 

**Similar issues observed in the following:**

- In System Default/Light theme,

1. visible focus is failing for "Filter", "Collapse All", "Refresh All" element present in explorer pane.
2. visible focus for "Edit field" of Blob Container Creation is failing minimum color contrast present in explorer pane.
3. Click on any Blob container, Toolbar menu is visible. All the tools present in the Toolbar is failing color contrast for visible focus.
4. Click on any Disk, Toolbar menu is visible. All the tools present in the Toolbar is failing color contrast for visible focus.

- In Dark Theme

1. Click on any Blob container, Toolbar menu is visible. All the tools present in the Toolbar is failing color contrast for visible focus.
2. Click on any Disk, Toolbar menu is visible. All the tools present in the Toolbar is failing color contrast for visible focus.

## Expected results:
Visible focus should have a minimum of 3:1 color contrast ratio.

## User Impact:
Low vision Users might fail to acknowledge the focused element if it doesn't have minimum color contrast ratio.

## Attachments:

| # | Screenshot |
|---|------------|
| 1 | ![5731_A11y_AzureStorageExplorer_DefaultTheme_CC_Search](https://user-images.githubusercontent.com/105198219/168602544-000414a7-50fd-4c36-95f1-1d60424a6afc.png) |
| 2 | ![5731_SimilarIssue_FilterButton](https://user-images.githubusercontent.com/105198219/168601438-cef43bfd-aaad-4633-a173-57a2a2aad963.png) |
| 3 | ![5731_SimilarIssue_CollapseAll](https://user-images.githubusercontent.com/105198219/168601424-d0dd6c47-3966-4d81-8674-f8b00eae98cb.png) |
| 4 | ![5731_SimilarIssues_RefreshAll](https://user-images.githubusercontent.com/105198219/168601442-079a529c-6371-4f2c-a8c0-d0578707b6b5.png) |
| 5 | ![5731_DefaultTheme_BlobContainer_Toolbar](https://user-images.githubusercontent.com/105198219/168602551-0a549997-f996-47fe-b16e-2626469bc2b8.png) |
| 6 | ![5731_DefaultTheme_diskToolbar](https://user-images.githubusercontent.com/105198219/168601767-0c9d975f-dab1-4fc9-a98b-d1b99c184a18.png) |
| 7 | ![5731_DarkTheme_BlobContainerToolbar](https://user-images.githubusercontent.com/105198219/168604163-3cb08282-92fd-4212-8f54-ad405323b010.png) |
| 8 | ![5731_DarkTheme_DiskToolbar](https://user-images.githubusercontent.com/105198219/168604174-ef96dd03-c63d-4c28-b4bc-b6bbc846ef14.png) |

**Comment by craxal** [comment]
@ManishaC1 I've merged fixes for screenshots 1-4. I'm still working on screenshots 5-8.

**Comment by craxal** [comment]
Changes have been merged, but packages still need to be updated to pick up the toolbar changes.

**Comment by MRayermannMSFT** [comment]
Hi @ManishaC1 

Please verify that this bug is fixed using the Storage Explorer build at: https://storageexplorerpublish.blob.core.windows.net/privatebuilds/20220624.3-accessibility-testing/StorageExplorer-ia32.exe

**Comment by ManishaC1** [comment]
Except one scenario, rest all scenarios have minimum Color contrast ratio.

Color contrast issue resolved for the scenarios shown in the attachments.

![5731_search button_1](https://user-images.githubusercontent.com/105198219/175870497-0a835007-0117-4b60-a3b9-8635ff7e66ac.png)
![5731_fiter button_2](https://user-images.githubusercontent.com/105198219/175870490-da0ab6d0-61aa-4ddc-bc75-53d38df382ab.png)
![5731_Blob container toolbar_3](https://user-images.githubusercontent.com/105198219/175870477-8ba366f1-2716-4022-a735-7e86770f6fd6.png)
![5731_Tool_Focus Indicator_4](https://user-images.githubusercontent.com/105198219/175870499-f7d3d345-15ae-44d6-affd-81915c34bcf7.png)
![5731_Dark theme_Toolbar](https://user-images.githubusercontent.com/105198219/175870484-821983b3-547d-44fc-bf3c-84b5f046c0f8.png)

Color contrast is failing for the scenario shown in the attachment. In dark theme, Focus indicator for all the disk tools have color contrast less than 3:1.
![5731_Issue Repro_dark theme](https://user-images.githubusercontent.com/105198219/175870870-751c5092-149f-445d-a92a-0d3345721b36.png)

**Comment by craxal** [comment]
For the color that's failing, according to the dev tools windown, it should be #007ACC. The color #0776AC is not defined or used in code anywhere that I can tell, so not sure where that color is coming from. I'll need to verify on my Windows machine.

**Comment by craxal** [comment]
@ManishaC1 I am unable to reproduce this issue. The focus color I am reading is #007ACC, which is the color we have coded for toolbar buttons, and it has a contrast ratio of 3.395:1. Can you please check this again?

Also, can you be more specific about how you sampled the color? Be advised, the eye dropper tool is highly dependent on where you sample the color and is not the best tool for gathering correct color information unless there is a solid block of pixels with the same color.

**Comment by ManishaC1** [comment]
@craxal I couldn't capture #007ACC color. How i sampled the color is that, usually we take the screenshot of the visible focus indicator and then we use color contrast because we cannot use color contrast analyzer just like that for focus indicator.

**Comment by craxal** [comment]
@ManishaC1 I don't think taking a screenshot and sampling the color is a great method, because the results may be wildly inconsistent due to a number of factors:
* App's zoom level, size of the color sample, antialiasing, blending, etc.
* Monitor resolution
* App color space, system color space
* Screenshot image format and compression

Just as an example, I took a screenshot myself and pasted it into Paint. I sampled the border (where the color is most consistent) and got #2A7BCF (which has at least 3:1 contrast). I also sampled your screenshot and I got #276EB5 (which is not 3:1, but more than the color you reported). Obviously neither of these matches #007ACC or #0769AC.

One way you can improve you sampling is to increase the app's zoom level to 200% or 300%. But the way I usually sample my colors is to use the Chromium developer window (pressing F12, or right clicking a tab and selecting "Open Developer Window"). This lets you select elements and force certain visual states, such as `:focus` or `:hover`. This will allow you to inspect the "intended" colors. Sadly, this is only possible in dev builds, not production builds.

We may need to bring up this issue in Accessibility office hours and find out how to reconcile these different colors. If we are expected to check everything against Accessibly Insights for Windows, then we have a serious engineering problem...

**Comment by craxal** [comment]
@ManishaC1 On second thought, never mind. It looks like the managed disks explorer wasn't using the latest toolbar version. I've updated that package, so you should be able to verify again using the next available build.

**Comment by ManishaC1** [comment]
@craxal Please let us know when the next build is released.

**Comment by ManishaC1** [comment]
Focus indicator for the disk tools have color contrast more than 3:1. Proof attached below. Since all the scenarios issue mentioned in this bug is resolved, thus closing the bug.
![Screenshot (525)](https://user-images.githubusercontent.com/105198219/178412659-e9676c83-b1b6-4474-bcce-6d60afdda33a.png)

**Comment by ManishaC1** [comment]
Github tags:#Closed

**Comment by Vishnuvarthancr7** [comment]
GitHubTags:#Closed;
